java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/NbCodeLanguageClient.java

```diff
@@ -69,6 +75,15 @@ public interface NbCodeLanguageClient extends LanguageClient {
     @JsonRequest("window/showInputBox")
     public CompletableFuture<String> showInputBox(@NonNull ShowInputBoxParams params);
 
+    /**
+     * Shows a mutli-step input using QuickPicks and InputBoxes.
+     *
+     * @param params input parameters
+     * @return collected input values and selected items
+     */
+    @JsonRequest("window/showMultiStepInput")
+    public CompletableFuture<Map<String, Either<List<QuickPickItem>, String>>> showMultiStepInput(@NonNull ShowMutliStepInputParams params);
```

Comment:
Using of `Either` is the way how union types in LSP interfaces are mapped to Java in LSP4J. The `NbCodeLanguageClient` class specifies how JSON based RPC messages between client and server look like, it does not represent an API exposed to the rest of NB. If there is a need to change it in the future, since it is local to the `java.lsp.server` module, it can be easily done in the similarly to any other change in LSP protocol.